sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/PodInfoBuilder.java

```diff
@@ -194,25 +193,26 @@ public void setExecutorVolume(VolumeSpec volumeSpec) {
 
     public static Protos.Resource getExistingExecutorVolume(
             VolumeSpec volumeSpec, String resourceId, String persistenceId) {
-        return Protos.Resource.newBuilder()
+        Protos.Resource.Builder resourceBuilder = Protos.Resource.newBuilder()
                 .setName("disk")
                 .setType(Protos.Value.Type.SCALAR)
-                .setScalar(volumeSpec.getValue().getScalar())
-                .setDisk(Protos.Resource.DiskInfo.newBuilder()
-                        .setPersistence(Protos.Resource.DiskInfo.Persistence.newBuilder()
-                                .setId(persistenceId)
-                                .setPrincipal(volumeSpec.getPrincipal()))
-                        .setVolume(Protos.Volume.newBuilder()
-                                .setContainerPath(volumeSpec.getContainerPath())
-                                .setMode(Protos.Volume.Mode.RW)))
-                .addReservations(Protos.Resource.ReservationInfo.newBuilder()
-                    .setPrincipal(volumeSpec.getPrincipal())
-                    .setRole(volumeSpec.getRole())
-                    .setLabels(Protos.Labels.newBuilder().addLabels(
-                            Protos.Label.newBuilder()
-                                    .setKey(MesosResource.RESOURCE_ID_KEY)
-                                    .setValue(resourceId))))
-                .build();
+                .setScalar(volumeSpec.getValue().getScalar());
+
+        Protos.Resource.DiskInfo.Builder diskInfoBuilder = resourceBuilder.getDiskBuilder();
+        diskInfoBuilder.getPersistenceBuilder()
+                .setId(persistenceId)
+                .setPrincipal(volumeSpec.getPrincipal());
+        diskInfoBuilder.getVolumeBuilder()
+                .setContainerPath(volumeSpec.getContainerPath())
+                .setMode(Protos.Volume.Mode.RW);
+
+        Protos.Resource.ReservationInfo.Builder reservationBuilder = resourceBuilder.addReservationsBuilder();
```

Comment:
L201-L207 are filling in `resource.disk` by grabbing the mutable Disk.Builder from resourceBuilder